test/dynamo/test_traceable_collectives.py

```diff
@@ -12,6 +12,11 @@ def dummy_collective_fn(input):
   return res_tensor
 
 
+def collective_broadcast_and_cos(input, src):
+  res_tensor = torch.ops._c10d_functional.broadcast(input, src, "default")
```

Comment:
Does it make more sense to test the Python API? https://github.com/pytorch/pytorch/blob/32c57e78edc46aa71ed19e013741c65b3d777fe9/torch/distributed/_functional_collectives.py#L143-L154

These functions (rather, the `_inplace versions`) are what Dynamo actually inserts as a replacement for the legacy `dist.*` collectives.

Reply:
so I tried a couple things, first I need to init the process group, but then 
```
res_tensor = torch.distributed.broadcast(input, src)
```
return None, I am guessing I need to pass group and tag correctly. Furthermore when I turn on `torch.compile` it failed with 
```
  File "/workspaces/dk4/pytorch/xla/torch_xla/_internal/pjrt.py", line 174, in run_multiprocess
    replica_results = list(
  File "/workspaces/dk4/pytorch/xla/torch_xla/_internal/pjrt.py", line 175, in <genexpr>
    itertools.chain.from_iterable(
  File "/usr/local/lib/python3.10/concurrent/futures/process.py", line 575, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 621, in result_iterator
    yield _result_or_cancel(fs.pop())
  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 319, in _result_or_cancel
    return fut.result(timeout)
  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
AssertionError: expected FunctionType found instancemethod <instancemethod  at 0x7f8d8456bf10>

from user code:
   File "/workspaces/dk4/pytorch/xla/torch_xla/distributed/xla_backend.py", line 98, in broadcast
    root_tensor = tensors[opts.rootTensor]
```

I think it got routed to https://github.com/pytorch/xla/blob/master/torch_xla/distributed/xla_backend.py#L97-L104 instead of my functional version. I will keep test as this to unblock VLLM for now.